app/services/iiif_print/derivative_rodeo_service.rb

```diff
@@ -0,0 +1,128 @@
+    # tomorrow wish.
+    class_attribute(:named_derivatives_and_generators_by_type, default: {
+                      pdf: { thumbnail: "DerivativeRodeo::Generators::ThumbnailGenerator" }
+                    })
+    # @!endgroup Class Attributes
+    ##
+
+    ##
+    # This method "hard-codes" some existing assumptions about the input_uri based on
+    # implementations for Adventist.  Those are reasonable assumptions but time will tell how
+    # reasonable.
+    #
+    # @param file_set [FileSet]
+    # @return [String]
+    def self.input_uri_for(file_set:)
+      # TODO: This is duplicated logic for another service, consider extracting a helper module;
+      # better yet wouldn't it be nice if Hyrax did this right and proper.
+      parent = file_set.parent || file_set.member_of.find(&:work?)
+      raise IiifPrint::DataError, "Parent not found for #{file_set.class} ID=#{file_set.id}" unless parent
+
+      dirname = parent.public_send(parent_work_identifier_property)
+
+      # TODO: This is a hack that knows about the inner workings of Hydra::Works, but for
+      # expendiency, I'm using it.  See
+      # https://github.com/samvera/hydra-works/blob/c9b9dd0cf11de671920ba0a7161db68ccf9b7f6d/lib/hydra/works/services/add_file_to_file_set.rb#L49-L53
+      # TODO: Could we get away with filename that is passed in the create_derivatives process?
+      filename = Hydra::Works::DetermineOriginalName.call(file_set.original_file)
+
+      # TODO: What kinds of exceptions might we raise if the location is not configured?  Do we need
+      # to "validate" it in another step.
+      location = DerivativeRodeo::StorageLocations::BaseLocation.load_location(input_location_adapter_name)
+
+      # TODO: This is based on the provided output template in
+      # https://github.com/scientist-softserv/space_stone-serverless/blob/0dbe2b6fa13b9f4bf8b9580ec14d0af5c98e2b00/awslambda/bin/sample_post.rb#L1
+      # and is very much hard-coded.  We likely want to "store" the template in a common place for
+      # the application.
+      File.join(location.adapter_prefix, dirname, filename)
+    end
+
+    def initialize(file_set)
+      @file_set = file_set
+    end
+
+    attr_reader :file_set
+    delegate :uri, :mime_type, to: :file_set
+
+    ##
+    # @return
+    # @see https://github.com/samvera/hyrax/blob/426575a9065a5dd3b30f458f5589a0a705ad7be2/app/services/hyrax/file_set_derivatives_service.rb#L18-L20 Hyrax::FileSetDerivativesService#valid?
+    def valid?
+      if in_the_rodeo?
+        Rails.logger.info("🤠🐮 Using the DerivativeRodeo for FileSet ID=#{file_set.id} with mime_type of #{mime_type}")
+        true
+      else
+        Rails.logger.info("Skipping the DerivativeRodeo for FileSet ID=#{file_set.id} with mime_type of #{mime_type}")
+        false
+      end
+    end
+
+    ##
+    # @api public
+    def create_derivatives(filename)
+      # TODO: Do we need to handle "impending derivatives?"  as per {IiifPrint::PluggableDerivativeService}?
```

Comment:
It seems like we might want to check valid? prior to actually generating the derivatives.

Reply:
The `Hyrax::DerivativeService` checks which registered service is valid and then uses that registered service.  So by the time we get to this method, it should be `valid?`